runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -727,12 +727,14 @@ local function add_injection(t, tree_index, pattern, lang, combined, ranges)
 end
 
 -- TODO(clason): replace by refactored `ts.has_parser` API (without registering)
----@param lang string parser name
----@return boolean # true if parser for {lang} exists on rtp
-local has_parser = function(lang)
+--- The result of this function is cached to prevent nvim_get_runtime_file from being
+--- called too often
+--- @param lang string parser name
+--- @return boolean # true if parser for {lang} exists on rtp
+local has_parser = vim._memoize(1, function(lang)
```

Comment:
I worry about this not being robust with respect to lazy-loading shenanigans... Maybe using `rtp` would be a reasonable hash function?

Even then, this will not pick up parsers installed with `:TSInstall`, which seems to be a definite regression.

(Not requesting changes, as this fixes an unacceptable performance regression, which trumps convenience, but wanted to make a note of it.)

Reply:
That's why we use a weak table. The cache will be invalidated when lua runs garbage collection which is  often enough.

If we use a normal table, then invalidation as you describe is required.

Reply:
I see. May be useful to mention in the memoize docs (or here), as I feel this will be a recurring question?

Reply:
There are only two hard things in Computer Science :)

Reply:
I already added: 

> Internally uses a weak table to cache the results of {fn}.

If people do not know what a weak table is then they can see `:h lua-weaktable`.

Reply:
Then maybe make this a link in the docs.